bleak/backends/bluezdbus/client.py

```diff
@@ -36,20 +37,23 @@ class BleakClientBlueZDBus(BaseBleakClient):
     Implemented by using the `BlueZ DBUS API <https://docs.ubuntu.com/core/en/stacks/bluetooth/bluez/docs/reference/dbus-api>`_.
 
     Args:
-        address (str): The  address of the BLE peripheral to connect to.
+        address_or_device (`BLEDevice` or str): The Bluetooth address of the BLE peripheral to connect to or the `BLEDevice` object representing it.
 
     Keyword Args:
         timeout (float): Timeout for required ``BleakScanner.find_device_by_address`` call. Defaults to 10.0.
 
     """
 
-    def __init__(self, address, **kwargs):
-        super(BleakClientBlueZDBus, self).__init__(address, **kwargs)
+    def __init__(self, address_or_device: Union[BLEDevice, str], **kwargs):
+        super(BleakClientBlueZDBus, self).__init__(address_or_device, **kwargs)
         self.device = kwargs.get("device") if kwargs.get("device") else "hci0"
```

Comment:
There are two different objects named `device` here, one is a BLEDevice, and the other is the BLE adapter name. I was confused by this at first: I didn't even know that you could pass the adapter name as an optional kwarg.

I'd suggest either renaming the argument to `address_or_ble_device`, or, better, I think, renaming the kwarg to `adapter` or something like that. It looks like the kwarg is only present in the bluez backend, so that incompatible change would not be widespread.

Reply:
Hmm.. Right you are. That is not a very pleasant clash...

I disliked the `address_or_device` argument name as well and hoped that a better one would come up in code review. I will think about it for a while. Maybe something closer to `peripheral`...

Reply:
You could split the functionality into a regular constructor and a class method. One would take an address, the other would take a device.

Upward compatible:
`BleakClient(address: str)`
`BleakClient.from_device(device: BLEDevice)`

Breaking:
`BleakClient(device: BLEDevice)`
`BleakClient.from_address(address: str)`

In most of the code I've written (not in bleak), the second case is the most common. I do a filtering scan looking for a device advertising a specific service or having a specific name. I don't know the address in advance.

Reply:
> Breaking:
> `BleakClient(device: BLEDevice)`
> `BleakClient.from_address(address: str)`

Other than the fact that this is breaking, I like this. Although we could just leave out the `from_address()` method since there is already `BleakScanner.find_device_by_address()`. ("There should be one-- and preferably only one --obvious way to do it.")

Reply:
I usually like to write classmethods for these kinds of special inits, but here I think it makes the basic usage more complex in a non-productive way. I would like to keep the init of a BleakClient as simple as possible, and rather accept two widely different kind of inputs to avoid confusing people. I get too many mails requesting that I write Bleak code as it is...

I plan on this to be very thouroughly documented in #266.